docs/manual/src/udl/callback_interfaces.md

```diff
@@ -1,107 +1,55 @@
 # Callback interfaces
 
-Callback interfaces are traits specified in UDL which can be implemented by foreign languages.
+Callback interfaces are traits specified in UDL which can be implemented by
+foreign languages.  Callbacks interfaces allow the library consumer to supply
+the Rust library with additional capabilities implemented in the foreign
+language, for example:
 
-They can provide Rust code access available to the host language, but not easily replicated
-in Rust.
+ * accessing app preferences and assets
+ * network requests
+ * logging
+ * task scheduling
 
- * accessing device APIs.
- * provide glue to clip together Rust components at runtime.
- * access shared resources and assets bundled with the app.
+Both sync and async functions are supported.
 
-# Using callback interfaces
-
-## 1. Define a Rust trait
-
-This toy example defines a way of Rust accessing a key-value store exposed
-by the host operating system (e.g. the key chain).
+# Defining callback interfaces
 
 ```rust,no_run
-pub trait Keychain: Send + Sync + Debug {
+#[uniffi::callback_interface]
```

Comment:
I think that if we want async we should definitely lean into the macros.  Rust currently doesn't allow async functions in traits, but a macro can convert those to a concrete type to work around it.  We could also support this in UDL, but then you'd need to write some slightly ugly Rust code -- async functions would need to be re-written as sync functions that returned `uniffi::ForeignFuture<T>`.